hydromt/data_sources/data_source.py

```diff
@@ -0,0 +1,53 @@
+"""Abstract DataSource class."""
+from typing import Any
+
+from pydantic import BaseModel, Field
+
+from hydromt import DataCatalog
+from hydromt.metadata_resolvers.resolver_plugin import RESOLVERS
+
+
+class DataSource(BaseModel):
```

Comment:
Not that this is wrong, but I think we should make a glossary of the terms used in the code because it's getting a bit hard for me to keep track of, let alone new folk. Here I mean the precise definition of things like source/catalog_item/adapter etc.

Reply:
Agreed. We can change terms as well